src/ServiceInsight/ServiceControl/ServiceControlConnectionProvider.cs

```diff
@@ -15,6 +15,11 @@ public void ConnectTo(string url)
         {
             Url = url;
 
+            if (url == null)
```

Comment:
Would it be better to throw an exception in this case? I guess we're fixing the problem at the call site below?